pkg/proxy/healthcheck/healthcheck.go

```diff
@@ -0,0 +1,291 @@
+/*
```

Comment:
please add a doc.go in this package that describes what happens when a new service is added, new endpoints are created etc. Eg: https://github.com/kubernetes/kubernetes/blob/master/pkg/dns/doc.go#L17, https://github.com/kubernetes/contrib/blob/master/ingress/controllers/gce/controller/doc.go. You're also free to just add it as a block comment at the head of this file, if you prefer.